tools/src/wyvern/tools/typedAST/core/expressions/Application.java

```diff
@@ -141,22 +160,31 @@ public FileLocation getLocation() {
 	public Expression generateIL(GenContext ctx, ValueType expectedType) {
 		CallableExprGenerator exprGen = function.getCallableExpr(ctx);
 		List<FormalArg> formals = exprGen.getExpectedArgTypes(ctx);
-		
+
+        int offset = 0;
 		// generate arguments		
 		List<Expression> args = new LinkedList<Expression>();
+        if(!this.generics.isEmpty()) {
+            for(String generic : this.generics) {
+                 args.add(new wyvern.target.corewyvernIL.expression.New(new TypeDeclaration(DefDeclaration.GENERIC_MEMBER, new NominalType("", generic), this.location)));
```

Comment:
Getting the formal at the index won't work if the function wasn't declared using `def [...]()`.